### ISSUE TITLE
Metrics Refactoring Corner Case

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#2612]
## 🐛 Bug

The recent metrics refactor has some corner cases where code that used to run before now breaks causing downstream user code errors.

## To Reproduce

To create a metric for precision recall, you could previously use

```
metric = dc.metrics.Metric(sklearn.metrics.precision_recall_curve)
```

Under the new metrics handling code, we now need to specify

```
metric = dc.metrics.Metric(sklearn.metrics.precision_recall_curve, mode="classification", classification_handling_mode="direct")
```

## Expected behavior

I'm not yet sure whether this is a bug or expected behavior. The new code is more precise about what's going on for sure but we should avoid causing downstream breakages in user code if possible here. Is there any way to improve our defaults so that the previous code sample runs correctly?

CC @peastman

**Comment by peastman** [comment]
The constructor for Metric sets defaults for standard metric functions, but it doesn't have anything for `precision_recall_curve`.  Should the defaults be "classification" and "direct"?
